File: src/main/java/um/haberes/core/model/view/CargoLiquidacionPeriodo.java
package um.haberes.core.model.view;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.Immutable;
import um.haberes.core.model.AuditableEntity;
import um.haberes.core.hexagonal.liquidaciones.categoria.infrastructure.persistence.entity.CategoriaEntity;
import um.haberes.core.hexagonal.personas.dependencia.infrastructure.persistence.entity.DependenciaEntity;
import um.haberes.core.hexagonal.personas.persona.infrastructure.persistence.entity.PersonaEntity;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Entity
@Table(name = "vw_cargo_liquidacion_periodo")
@Immutable
@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class CargoLiquidacionPeriodo extends AuditableEntity {

    @Id
    private Long cargoLiquidacionId = null;

    private Long legajoId = null;

    private Integer anho = null;

    private Integer mes = null;

    private Integer dependenciaId = null;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaDesde = null;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaHasta = null;

    private Integer categoriaId = null;

    private String categoriaNombre = null;

    private BigDecimal categoriaBasico = null;

    private int jornada = 0;

    private int presentismo = 0;

    private BigDecimal asignacionEspecialPermanente = BigDecimal.ZERO;

    private String situacion = null;

    private Long periodo = null;

    @OneToOne(optional = true)
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    private PersonaEntity persona = null;

    @OneToOne(optional = true)
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    private DependenciaEntity dependencia = null;

    @OneToOne(optional = true)
    @JoinColumn(name = "categoriaId", insertable = false, updatable = false)
    private CategoriaEntity categoria = null;
}

File: src/main/java/um/haberes/core/kotlin/model/view/CargoLiquidacionPeriodo.kt
package um.haberes.core.kotlin.model.view

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import org.hibernate.annotations.Immutable
import um.haberes.core.kotlin.model.Auditable
import um.haberes.core.kotlin.model.Categoria
import um.haberes.core.kotlin.model.Dependencia
import um.haberes.core.kotlin.model.Persona
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(name = "vw_cargo_liquidacion_periodo")
@Immutable
data class CargoLiquidacionPeriodo(

    @Id
    var cargoLiquidacionId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int? = null,
    var mes: Int? = null,
    var dependenciaId: Int? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaDesde: OffsetDateTime? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaHasta: OffsetDateTime? = null,

    var categoriaId: Int? = null,
    var categoriaNombre: String? = null,
    var categoriaBasico: BigDecimal? = null,
    var jornada: Int = 0,
    var presentismo: Int = 0,
    var asignacionEspecialPermanente: BigDecimal = BigDecimal.ZERO,
    var situacion: String? = null,
    var periodo: Long? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    var dependencia: Dependencia? = null,

    @OneToOne(optional = true)
    @JoinColumn(name = "categoriaId", insertable = false, updatable = false)
    var categoria: Categoria? = null

) : Auditable()
